src/foam/core/EmptyX.java

```diff
@@ -124,4 +193,15 @@ private EmptyX() {}
   public static X instance() { return x_; }
 
   public Object get(X x, Object key) { return null; }
+
+  public X put(Object key, Object value) {
+    return new XI(this, this, key, value);
```

Comment:
Is this correct?

Reply:
When context itself is EmptyX, or context can not find key until reaches the EmptyX, the EmptyX will directly create a XI or FactoryXI and make all children pointer point EmptyX, and return the reference to its parent. The cloned parent will hold the reference to access this child. In this case, all children pointers of the leafs will point to the EmptyX.